node/lib/util/synthetic_branch_util.js

```diff
@@ -96,6 +96,29 @@ exports.urlToLocalPath = function *(repo, url) {
     }
 };
 
+/**
+ * Check that a given URL i son the URLs synthetic-ref-check whitelist, if
+ * such a whitelist exists.
+ * @async
+ * @param {NodeGit.Repository} repo The meta repository
+ * @param {String}             url The configured URL of hte submodule
+ * in the meta tree.
+ */
+function* skipCheckForURL(repo, url) {
+    assert.instanceOf(repo, NodeGit.Repository);
+    assert.isString(url);
+    const config = yield repo.config();
```

Comment:
I am worried about doing this on every submodule url because I'm not sure it's cached -- could we extract it?